src/Button.tsx

```diff
@@ -79,37 +72,23 @@ const defaultProps = {
 
 const Button: BsPrefixRefForwardingComponent<'button', ButtonProps> =
   React.forwardRef<HTMLButtonElement, ButtonProps>(
-    (
-      { bsPrefix, variant, size, active, className, type, as, ...props },
-      ref,
-    ) => {
+    ({ bsPrefix, variant, size, active, className, ...props }, ref) => {
       const prefix = useBootstrapPrefix(bsPrefix, 'btn');
 
-      const classes = classNames(
-        className,
-        prefix,
-        active && 'active',
-        variant && `${prefix}-${variant}`,
-        size && `${prefix}-${size}`,
+      return (
```

Comment:
might be worth using the hook directly? avoids an extra component for a little more logic